frontend/src/app/shared/utils/yup.ts

```diff
@@ -57,13 +61,25 @@ function atLeastOneStringI18n(this: Yup.ObjectSchema<any>) {
         });
 }
 
+function formatI18n(this: Yup.StringSchema, format: PropertyFormat) {
```

Comment:
I would not make this. It creates a very weird dependency from a utility class to the domain layer (service). You only need this method in one place anyway, so it does not reduces the amount of code.